### PR TITLE
test: add Jwt, SM2Encryption and SM3Password util test.

### PR DESCRIPTION
**Pull request by msslulu**
feat: jwtutil,SM2EncryptionUtil,SM3PasswordUtil UT

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced input validation in authentication and cryptographic utilities to reject null parameters and prevent potential runtime errors.

* **Tests**
  * Added comprehensive test suites covering authentication token handling, encryption/decryption workflows, and password hashing with edge cases, malformed inputs, and concurrency scenarios.

### CODE REVIEW

**Review by chilingling — DISMISSED** [review]

File: base/src/main/java/com/tinyengine/it/login/utils/SM3PasswordUtil.java [hunk 1]
```diff
@@ -54,6 +57,9 @@ public static PasswordResult createPassword(String plainPassword) throws Excepti
      * 验证用户密码
      */
     public static boolean verifyPassword(String inputPassword, String storedHash, String salt) throws Exception {
+        if(inputPassword == null || storedHash == null || salt == null) {
+            throw new IllegalArgumentException("输入密码、存储哈希和盐值不能为空");
+        }
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/main/java/com/tinyengine/it/login/utils/SM3PasswordUtil.java:62]: no need change

File: base/src/test/java/com/tinyengine/it/login/utils/JwtUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+		when(environment.getActiveProfiles()).thenReturn(new String[]{"dev"});
+		assertDoesNotThrow(jwtUtil::validateSecretConfiguration);
+	}
+
+
+	@Test
+	void validateSecretConfigurationThrowsExceptionForInvalidSecret() {
+		when(environment.getActiveProfiles()).thenReturn(new String[]{"prod"});
+		System.setProperty("SECRET_STRING", "short");
+		assertThrows(IllegalStateException.class, jwtUtil::validateSecretConfiguration);
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/test/java/com/tinyengine/it/login/utils/JwtUtilTest.java:64]: no need chnage

File: base/src/test/java/com/tinyengine/it/login/utils/JwtUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+	void validateTokenHandlesTokenWithInvalidAlgorithm() {
+		String token = jwtUtil.generateToken("testUser", "USER", "123", null, 1);
+		when(tokenBlacklistService.isTokenBlacklisted(token)).thenReturn(true);
+
+		// Simulate invalid algorithm
+		String tamperedToken = token.replace("HS256", "RS256");
+
+		boolean isValid = jwtUtil.validateToken(tamperedToken);
+
+		assertFalse(isValid, "Token with invalid algorithm should return false");
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/test/java/com/tinyengine/it/login/utils/JwtUtilTest.java:325]: no need change

File: base/src/test/java/com/tinyengine/it/login/utils/SM2EncryptionUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+			"Invalid Base64 key should throw InvalidKeySpecException");
+	}
+
+	@Test
+	void decryptWithInvalidCipherText() {
+		String invalidCipherText = "invalidCipherText";
+		PrivateKey privateKey = keyPair.getPrivate();
+
+		assertThrows(Exception.class, () -> SM2EncryptionUtil.decrypt(invalidCipherText, privateKey),
+			"Decrypting invalid cipher text should throw an exception");
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/test/java/com/tinyengine/it/login/utils/SM2EncryptionUtilTest.java:145]: no need change

File: base/src/test/java/com/tinyengine/it/login/utils/SM3PasswordUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,493 @@
...
+		// 先创建一个合法结果
+		assertThrows(Exception.class, () -> {
+			SM3PasswordUtil.verifyPassword(null, "hash", "salt");
+		});
+		assertThrows(Exception.class, () -> {
+			SM3PasswordUtil.verifyPassword("pwd", null, "salt");
+		});
+		assertThrows(Exception.class, () -> {
+			SM3PasswordUtil.verifyPassword("pwd", "hash", null);
+		});
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/test/java/com/tinyengine/it/login/utils/SM3PasswordUtilTest.java:176]: no need change

File: base/src/test/java/com/tinyengine/it/login/utils/SM3PasswordUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,493 @@
...
+				for (int i = 0; i < iterationsPerThread; i++) {
+					String pwd = "pwd" + Thread.currentThread().getId() + "_" + i;
+					PasswordResult result = SM3PasswordUtil.createPassword(pwd);
+					assertTrue(SM3PasswordUtil.verifyPassword(pwd, result.getPasswordHash(), result.getSalt()));
+				}
+			} catch (Exception e) {
+				synchronized (exceptions) {
+					exceptions[0] = e;
+				}
+			}
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/test/java/com/tinyengine/it/login/utils/SM3PasswordUtilTest.java:418]: no need change

File: base/src/test/java/com/tinyengine/it/login/utils/SM3PasswordUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,493 @@
...
+	@DisplayName("批量创建并验证 1000 个密码应在 2 秒内完成")
+	@Timeout(2)
+	void performance_BatchCreateAndVerify() throws Exception {
+		int iterations = 1000;
+		for (int i = 0; i < iterations; i++) {
+			String plain = "user" + i + "P@ss";
+			PasswordResult result = SM3PasswordUtil.createPassword(plain);
+			assertTrue(SM3PasswordUtil.verifyPassword(plain, result.getPasswordHash(), result.getSalt()));
+		}
+	}
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/test/java/com/tinyengine/it/login/utils/SM3PasswordUtilTest.java:275]: no need change

**Review by chilingling — APPROVED** [review]
